### ISSUE TITLE
bug: JDTLS crashes when opening up java file

### ISSUE THREAD

**Issue by exosyphon** [nvim-java/nvim-java#135]
### Did you check docs and existing issues?

- [X] I have read all the plugin docs
- [X] I have searched the existing issues
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.9.5

### Operating system/version

MacOS 14.4

### Describe the bug

Using a fresh install of Kickstart.nvim I add the example config and when I open a java file in an existing project then it crashes with this error:

```
[WARN][2024-03-15 21:46:29] ...lsp/handlers.lua:537	'Mar 15, 2024, 9:46:29 PM While loading class "org.eclipse.buildship.core.internal.preferences.PersistentModelConverter$12", thread "Thread[Worker-3: Load persistent model for all projects,5,main]" timed out waiting (30045ms) for thread "Thread[Worker-0: Processing Java changes since last activation,5,main]" to finish starting bundle "org.eclipse.buildship.core_3.1.8.v20231117-1658 [110]". To avoid deadlock, thread "Thread[Worker-3: Load persistent model for all projects,5,main]" is proceeding but "org.eclipse.buildship.core.internal.preferences.PersistentModelConverter$12" may not be fully initialized.\nUnable to acquire the state change lock for the module: osgi.identity; type="osgi.bundle"; version:Version="3.1.8.v20231117-1658"; osgi.identity="org.eclipse.buildship.core"; singleton:="true" [id=110] STARTED [STARTED]\norg.osgi.framework.BundleException: Unable to acquire the state change lock for the module: osgi.identity; type="osgi.bundle"; version:Version="3.1.8.v20231117-1658"; osgi.identity="org.eclipse.buildship.core"; singleton:="true" [id=110] STARTED [STARTED]\n\tat org.eclipse.osgi.container.Module.lockStateChange(Module.java:373)\n\tat org.eclipse.osgi.container.Module.start(Module.java:447)\n\tat org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:528)\n\tat org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:122)\n\tat org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:620)\n\tat org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:348)\n\tat org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:414)\n\tat org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:520)\n\tat org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:434)\n\tat org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:174)\n\tat java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)\n\tat org.eclipse.buildship.core.internal.preferences.PersistentModelConverter.toModel(PersistentModelConverter.java:142)\n\tat org.eclipse.buildship.core.internal.preferences.DefaultModelPersistence.doLoadModel(DefaultModelPersistence.java:130)\n\tat org.eclipse.buildship.core.internal.preferences.DefaultModelPersistence$1.load(DefaultModelPersistence.java:62)\n\tat org.eclipse.buildship.core.internal.preferences.DefaultModelPersistence$1.load(DefaultModelPersistence.java:57)\n\tat com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3576)\n\tat com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2318)\n\tat com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2191)\n\tat com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2081)\n\tat com.google.common.cache.LocalCache.get(LocalCache.java:4019)\n\tat com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4042)\n\tat com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5024)\n\tat org.eclipse.buildship.core.internal.preferences.DefaultModelPersistence$2.run(DefaultModelPersistence.java:193)\n\tat org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)\nCaused by: java.util.concurrent.TimeoutException: Timeout after waiting 30 seconds to acquire the lock.\n\tat org.eclipse.osgi.container.Module.lockStateChange(Module.java:368)\n\t... 23 more\nCaused by: org.eclipse.osgi.framework.util.ThreadInfoReport: Thread dump\n\nThreadId: 1 ThreadName: main ThreadState: WAITING\n  Blocked On: java.lang.Object@5ad5abcd LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Object.wait(Native Method)\n    java.base@17.0.10/java.lang.Object.wait(Object.java:338)\n    org.eclipse.jdt.ls.core.internal.LanguageServerApplication.start(LanguageServerApplication.java:70)\n    org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)\n    org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)\n    org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)\n    org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)\n    org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)\n    java.base@17.0.10/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n    java.base@17.0.10/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)\n    java.base@17.0.10/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n    java.base@17.0.10/java.lang.reflect.Method.invoke(Method.java:568)\n    app//org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:651)\n    app//org.eclipse.equinox.launcher.Main.basicRun(Main.java:588)\n    app//org.eclipse.equinox.launcher.Main.run(Main.java:1459)\n    app//org.eclipse.equinox.launcher.Main.main(Main.java:1432)\n\nThreadId: 2 ThreadName: Reference Handler ThreadState: RUNNABLE\n  Blocked On: none\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.ref.Reference.waitForReferencePendingList(Native Method)\n    java.base@17.0.10/java.lang.ref.Reference.processPendingReferences(Reference.java:253)\n    java.base@17.0.10/java.lang.ref.Reference$ReferenceHandler.run(Reference.java:215)\n\nThreadId: 3 ThreadName: Finalizer ThreadState: WAITING\n  Blocked On: java.lang.ref.ReferenceQueue$Lock@7821d4f7 LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Object.wait(Native Method)\n    java.base@17.0.10/java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:155)\n    java.base@17.0.10/java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:176)\n    java.base@17.0.10/java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:172)\n\nThreadId: 4 ThreadName: Signal Dispatcher ThreadState: RUNNABLE\n  Blocked On: none\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n\nThreadId: 12 ThreadName: Common-Cleaner ThreadState: TIMED_WAITING\n  Blocked On: java.lang.ref.ReferenceQueue$Lock@1b062dad LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Object.wait(Native Method)\n    java.base@17.0.10/java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:155)\n    java.base@17.0.10/jdk.internal.ref.CleanerImpl.run(CleanerImpl.java:140)\n    java.base@17.0.10/java.lang.Thread.run(Thread.java:840)\n    java.base@17.0.10/jdk.internal.misc.InnocuousThread.run(InnocuousThread.java:162)\n\nThreadId: 13 ThreadName: Notification Thread ThreadState: RUNNABLE\n  Blocked On: none\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n\nThreadId: 16 ThreadName: Active Thread: Equinox Container: 3cc47dad-d8a7-4058-aeb6-ae9c7d860340 ThreadState: TIMED_WAITING\n  Blocked On: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@27ef8fe8 LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/jdk.internal.misc.Unsafe.park(Native Method)\n    java.base@17.0.10/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)\n    java.base@17.0.10/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:1672)\n    java.base@17.0.10/java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1182)\n    java.base@17.0.10/java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:899)\n    java.base@17.0.10/java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1062)\n    java.base@17.0.10/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1122)\n    java.base@17.0.10/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n    java.base@17.0.10/java.lang.Thread.run(Thread.java:840)\n\nThreadId: 18 ThreadName: Framework Event Dispatcher: Equinox Container: 3cc47dad-d8a7-4058-aeb6-ae9c7d860340 ThreadState: WAITING\n  Blocked On: org.eclipse.osgi.framework.eventmgr.EventManager$EventThread@5cd6907d LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Object.wait(Native Method)\n    java.base@17.0.10/java.lang.Object.wait(Object.java:338)\n    org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:396)\n    org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:337)\n\nThreadId: 19 ThreadName: Start Level: Equinox Container: 3cc47dad-d8a7-4058-aeb6-ae9c7d860340 ThreadState: WAITING\n  Blocked On: org.eclipse.osgi.framework.eventmgr.EventManager$EventThread@54157b7c LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Object.wait(Native Method)\n    java.base@17.0.10/java.lang.Object.wait(Object.java:338)\n    org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:396)\n    org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:337)\n\nThreadId: 21 ThreadName: Bundle File Closer ThreadState: WAITING\n  Blocked On: org.eclipse.osgi.framework.eventmgr.EventManager$EventThread@2460954 LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Object.wait(Native Method)\n    java.base@17.0.10/java.lang.Object.wait(Object.java:338)\n    org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:396)\n    org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:337)\n\nThreadId: 22 ThreadName: SCR Component Actor ThreadState: WAITING\n  Blocked On: java.util.LinkedList@2a0c2c36 LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Object.wait(Native Method)\n    java.base@17.0.10/java.lang.Object.wait(Object.java:338)\n    org.apache.felix.scr.impl.ComponentActorThread.run(ComponentActorThread.java:83)\n    java.base@17.0.10/java.lang.Thread.run(Thread.java:840)\n\nThreadId: 24 ThreadName: Worker-JM ThreadState: TIMED_WAITING\n  Blocked On: java.util.ArrayList@71c77650 LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Object.wait(Native Method)\n    org.eclipse.core.internal.jobs.InternalWorker.run(InternalWorker.java:61)\n\nThreadId: 25 ThreadName: Worker-0: Processing Java changes since last activation ThreadState: BLOCKED\n  Blocked On: java.lang.Object@30648e4d LockOwnerId: 29 LockOwnerName: Worker-3: Load persistent model for all projects\n  Synchronizers Locked: \n    java.util.concurrent.locks.ReentrantLock$NonfairSync@238ee4e5\n  Monitors Locked: \n    com.google.common.cache.LocalCache$StrongEntry@7f3f3e58\n  Stack Trace: \n    org.eclipse.buildship.core.internal.preferences.DefaultModelPersistence$1.load(DefaultModelPersistence.java:62)\n    org.eclipse.buildship.core.internal.preferences.DefaultModelPersistence$1.load(DefaultModelPersistence.java:57)\n    com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3576)\n    com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2318)\n    com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2191)\n    com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2081)\n    com.google.common.cache.LocalCache.get(LocalCache.java:4019)\n    com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4042)\n    com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5024)\n    com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:5031)\n    org.eclipse.buildship.core.internal.preferences.DefaultModelPersistence.loadModel(DefaultModelPersistence.java:70)\n    org.eclipse.buildship.core.internal.CorePlugin.scheduleSynchronizationForAbsentModels(CorePlugin.java:223)\n    org.eclipse.buildship.core.internal.CorePlugin.start(CorePlugin.java:120)\n    org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:833)\n    org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:1)\n    java.base@17.0.10/java.security.AccessController.executePrivileged(AccessController.java:807)\n    java.base@17.0.10/java.security.AccessController.doPrivileged(AccessController.java:569)\n    org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:825)\n    org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:775)\n    org.eclipse.osgi.internal.framework.EquinoxBundle.startWorker0(EquinoxBundle.java:1057)\n    org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.startWorker(EquinoxBundle.java:387)\n    org.eclipse.osgi.container.Module.doStart(Module.java:639)\n    org.eclipse.osgi.container.Module.start(Module.java:498)\n    org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:528)\n    org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:122)\n    org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:620)\n    org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:348)\n    org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:414)\n    org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:520)\n    org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:434)\n    org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:174)\n    java.base@17.0.10/java.lang.ClassLoader.loadClass(ClassLoader.java:525)\n    org.eclipse.osgi.internal.framework.EquinoxBundle.loadClass(EquinoxBundle.java:643)\n    org.eclipse.core.internal.registry.osgi.RegistryStrategyOSGI.createExecutableExtension(RegistryStrategyOSGI.java:223)\n    org.eclipse.core.internal.registry.ExtensionRegistry.createExecutableExtension(ExtensionRegistry.java:987)\n    org.eclipse.core.internal.registry.ConfigurationElement.createExecutableExtension(ConfigurationElement.java:255)\n    org.eclipse.core.internal.registry.ConfigurationElementHandle.createExecutableExtension(ConfigurationElementHandle.java:65)\n    org.eclipse.jdt.core.JavaCore.computeClasspathContainerInitializer(JavaCore.java:3986)\n    org.eclipse.jdt.core.JavaCore.getClasspathContainerInitializer(JavaCore.java:3962)\n    org.eclipse.jdt.internal.core.JavaModelManager.initializeContainer(JavaModelManager.java:3292)\n    org.eclipse.jdt.internal.core.JavaModelManager$11.run(JavaModelManager.java:3195)\n    org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2448)\n    org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2473)\n    org.eclipse.jdt.internal.core.JavaModelManager.initializeAllContainers(JavaModelManager.java:3257)\n    org.eclipse.jdt.internal.core.JavaModelManager.getClasspathContainer(JavaModelManager.java:2209)\n    org.eclipse.jdt.core.JavaCore.getClasspathContainer(JavaCore.java:3939)\n    org.eclipse.jdt.internal.core.JavaProject.resolveClasspath(JavaProject.java:3159)\n    org.eclipse.jdt.internal.core.JavaProject.resolveClasspath(JavaProject.java:3323)\n    org.eclipse.jdt.internal.core.ClasspathChange.generateDelta(ClasspathChange.java:249)\n    org.eclipse.jdt.internal.core.DeltaProcessor.resourceChanged(DeltaProcessor.java:2134)\n    org.eclipse.jdt.internal.core.DeltaProcessingState.resourceChanged(DeltaProcessingState.java:503)\n    org.eclipse.core.internal.events.NotificationManager$1.run(NotificationManager.java:321)\n    org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)\n    org.eclipse.core.internal.events.NotificationManager.notify(NotificationManager.java:311)\n    org.eclipse.core.internal.events.NotificationManager.broadcastChanges(NotificationManager.java:211)\n    org.eclipse.core.internal.resources.SavedState.processResourceChangeEvents(SavedState.java:88)\n    org.eclipse.jdt.internal.core.JavaModelManager$21$1.run(JavaModelManager.java:5568)\n    org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2448)\n    org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2468)\n    org.eclipse.jdt.internal.core.JavaModelManager$21.run(JavaModelManager.java:5559)\n    org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)\n\nThreadId: 26 ThreadName: Worker-1: Refreshing workspace ThreadState: WAITING\n  Blocked On: java.lang.Object@36afabe7 LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Object.wait(Native Method)\n    java.base@17.0.10/java.lang.Object.wait(Object.java:338)\n    org.eclipse.core.internal.jobs.ThreadJob.waitForRun(ThreadJob.java:335)\n    org.eclipse.core.internal.jobs.ThreadJob.joinRun(ThreadJob.java:214)\n    org.eclipse.core.internal.jobs.ImplicitJobs.begin(ImplicitJobs.java:95)\n    org.eclipse.core.internal.jobs.JobManager.beginRule(JobManager.java:341)\n    org.eclipse.core.internal.refresh.RefreshJob.runInWorkspace(RefreshJob.java:224)\n    org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:43)\n    org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)\n\nThreadId: 27 ThreadName: Worker-2 ThreadState: TIMED_WAITING\n  Blocked On: org.eclipse.core.internal.jobs.WorkerPool@41ce40fd LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Object.wait(Native Method)\n    org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)\n    org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)\n    org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)\n\nThreadId: 28 ThreadName: Java indexing ThreadState: WAITING\n  Blocked On: org.eclipse.jdt.internal.core.search.indexing.IndexManager@15b33ded LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Object.wait(Native Method)\n    java.base@17.0.10/java.lang.Object.wait(Object.java:338)\n    org.eclipse.jdt.internal.core.search.processing.JobManager.indexerLoop(JobManager.java:503)\n    org.eclipse.jdt.internal.core.search.processing.JobManager$$Lambda$214/0x000000080036cf30.run(Unknown Source)\n    java.base@17.0.10/java.lang.Thread.run(Thread.java:840)\n\nThreadId: 29 ThreadName: Worker-3: Load persistent model for all projects ThreadState: RUNNABLE\n  Blocked On: org.eclipse.osgi.internal.container.EquinoxReentrantLock@7e643a2c[Locked by thread Worker-0: Processing Java changes since last activation]\n  Synchronizers Locked: none\n  Monitors Locked: \n    java.lang.Object@30648e4d\n    com.google.common.cache.LocalCache$StrongEntry@131cff0\n  Stack Trace: \n    java.management@17.0.10/sun.management.ThreadImpl.dumpThreads0(Native Method)\n    java.management@17.0.10/sun.management.ThreadImpl.dumpAllThreads(ThreadImpl.java:521)\n    java.management@17.0.10/sun.management.ThreadImpl.dumpAllThreads(ThreadImpl.java:509)\n    org.eclipse.osgi.framework.util.ThreadInfoReport.getThreadDump(ThreadInfoReport.java:30)\n    org.eclipse.osgi.framework.util.ThreadInfoReport.<init>(ThreadInfoReport.java:23)\n    org.eclipse.osgi.container.Module.lockStateChange(Module.java:370)\n    org.eclipse.osgi.container.Module.start(Module.java:447)\n    org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:528)\n    org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:122)\n    org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:620)\n    org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:348)\n    org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:414)\n    org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:520)\n    org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:434)\n    org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:174)\n    java.base@17.0.10/java.lang.ClassLoader.loadClass(ClassLoader.java:525)\n    org.eclipse.buildship.core.internal.preferences.PersistentModelConverter.toModel(PersistentModelConverter.java:142)\n    org.eclipse.buildship.core.internal.preferences.DefaultModelPersistence.doLoadModel(DefaultModelPersistence.java:130)\n    org.eclipse.buildship.core.internal.preferences.DefaultModelPersistence$1.load(DefaultModelPersistence.java:62)\n    org.eclipse.buildship.core.internal.preferences.DefaultModelPersistence$1.load(DefaultModelPersistence.java:57)\n    com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3576)\n    com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2318)\n    com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2191)\n    com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2081)\n    com.google.common.cache.LocalCache.get(LocalCache.java:4019)\n    com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4042)\n    com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5024)\n    org.eclipse.buildship.core.internal.preferences.DefaultModelPersistence$2.run(DefaultModelPersistence.java:193)\n    org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)\n\nThreadId: 30 ThreadName: Worker-4 ThreadState: TIMED_WAITING\n  Blocked On: org.eclipse.core.internal.jobs.WorkerPool@41ce40fd LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Object.wait(Native Method)\n    org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)\n    org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)\n    org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)\n\nThreadId: 35 ThreadName: Worker-5: Updating Maven Dependencies ThreadState: WAITING\n  Blocked On: java.lang.Object@36afabe7 LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Object.wait(Native Method)\n    java.base@17.0.10/java.lang.Object.wait(Object.java:338)\n    org.eclipse.core.internal.jobs.ThreadJob.waitForRun(ThreadJob.java:335)\n    org.eclipse.core.internal.jobs.ThreadJob.joinRun(ThreadJob.java:214)\n    org.eclipse.core.internal.jobs.ImplicitJobs.begin(ImplicitJobs.java:95)\n    org.eclipse.core.internal.jobs.JobManager.beginRule(JobManager.java:341)\n    org.eclipse.m2e.core.internal.project.registry.ProjectRegistryRefreshJob.lambda$0(ProjectRegistryRefreshJob.java:160)\n    org.eclipse.m2e.core.internal.project.registry.ProjectRegistryRefreshJob$$Lambda$280/0x00000008002f6c48.call(Unknown Source)\n    org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:394)\n    org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:275)\n    org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:214)\n    org.eclipse.m2e.core.internal.project.registry.ProjectRegistryRefreshJob.run(ProjectRegistryRefreshJob.java:106)\n    org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)\n\nThreadId: 36 ThreadName: pool-2-thread-1 ThreadState: TIMED_WAITING\n  Blocked On: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@4a0abca2 LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/jdk.internal.misc.Unsafe.park(Native Method)\n    java.base@17.0.10/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)\n    java.base@17.0.10/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:1672)\n    java.base@17.0.10/java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1182)\n    java.base@17.0.10/java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:899)\n    java.base@17.0.10/java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1062)\n    java.base@17.0.10/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1122)\n    java.base@17.0.10/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n    java.base@17.0.10/java.lang.Thread.run(Thread.java:840)\n\nThreadId: 37 ThreadName: WorkspaceEventsHandler ThreadState: WAITING\n  Blocked On: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@7970d4b6 LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/jdk.internal.misc.Unsafe.park(Native Method)\n    java.base@17.0.10/java.util.concurrent.locks.LockSupport.park(LockSupport.java:341)\n    java.base@17.0.10/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode.block(AbstractQueuedSynchronizer.java:506)\n    java.base@17.0.10/java.util.concurrent.ForkJoinPool.unmanagedBlock(ForkJoinPool.java:3465)\n    java.base@17.0.10/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3436)\n    java.base@17.0.10/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:1623)\n    java.base@17.0.10/java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:435)\n    org.eclipse.jdt.ls.core.internal.handlers.WorkspaceEventsHandler.lambda$0(WorkspaceEventsHandler.java:60)\n    org.eclipse.jdt.ls.core.internal.handlers.WorkspaceEventsHandler$$Lambda$276/0x0000000800452000.run(Unknown Source)\n    java.base@17.0.10/java.lang.Thread.run(Thread.java:840)\n\nThreadId: 38 ThreadName: pool-1-thread-1 ThreadState: TIMED_WAITING\n  Blocked On: none\n  Synchronizers Locked: \n    java.util.concurrent.ThreadPoolExecutor$Worker@132ddbab\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Thread.sleep(Native Method)\n    org.eclipse.jdt.ls.core.internal.JobHelpers.waitForJobs(JobHelpers.java:198)\n    org.eclipse.jdt.ls.core.internal.JobHelpers.waitForProjectRegistryRefreshJob(JobHelpers.java:238)\n    org.eclipse.jdt.ls.core.internal.handlers.InitHandler.handleInitializationOptions(InitHandler.java:101)\n    org.eclipse.jdt.ls.core.internal.handlers.BaseInitHandler.initialize(BaseInitHandler.java:64)\n    org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.initialize(JDTLanguageServer.java:284)\n    java.base@17.0.10/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n    java.base@17.0.10/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)\n    java.base@17.0.10/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n    java.base@17.0.10/java.lang.reflect.Method.invoke(Method.java:568)\n    org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$recursiveFindRpcMethods$0(GenericEndpoint.java:65)\n    org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint$$Lambda$268/0x000000080042f5e8.apply(Unknown Source)\n    org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)\n    org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)\n    org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)\n    org.eclipse.jdt.ls.core.internal.ParentProcessWatcher.lambda$1(ParentProcessWatcher.java:144)\n    org.eclipse.jdt.ls.core.internal.ParentProcessWatcher$$Lambda$266/0x000000080043a180.consume(Unknown Source)\n    org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)\n    org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)\n    org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)\n    java.base@17.0.10/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)\n    java.base@17.0.10/java.util.concurrent.FutureTask.run(FutureTask.java:264)\n    java.base@17.0.10/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n    java.base@17.0.10/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n    java.base@17.0.10/java.lang.Thread.run(Thread.java:840)\n\nThreadId: 39 ThreadName: Worker-6: Updating workspace ThreadState: WAITING\n  Blocked On: com.google.common.util.concurrent.SettableFuture@4d94f0f5 LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/jdk.internal.misc.Unsafe.park(Native Method)\n    java.base@17.0.10/java.util.concurrent.locks.LockSupport.park(LockSupport.java:211)\n    com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:561)\n    com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:111)\n    com.google.common.util.concurrent.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:247)\n    com.google.common.cache.LocalCache$LoadingValueReference.waitForValue(LocalCache.java:3628)\n    com.google.common.cache.LocalCache$Segment.waitForLoadingValue(LocalCache.java:2211)\n    com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2198)\n    com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2081)\n    com.google.common.cache.LocalCache.get(LocalCache.java:4019)\n    com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4042)\n    com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5024)\n    com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:5031)\n    org.eclipse.buildship.core.internal.preferences.DefaultModelPersistence.loadModel(DefaultModelPersistence.java:70)\n    org.eclipse.buildship.core.internal.workspace.GradleClasspathContainerUpdater.updateFromStorage(GradleClasspathContainerUpdater.java:189)\n    org.eclipse.buildship.core.internal.workspace.GradleClasspathContainerInitializer.updateFromStorage(GradleClasspathContainerInitializer.java:68)\n    org.eclipse.buildship.core.internal.workspace.GradleClasspathContainerInitializer.loadClasspath(GradleClasspathContainerInitializer.java:51)\n    org.eclipse.buildship.core.internal.workspace.GradleClasspathContainerInitializer.initialize(GradleClasspathContainerInitializer.java:41)\n    org.eclipse.jdt.internal.core.JavaModelManager.initializeContainer(JavaModelManager.java:3311)\n    org.eclipse.jdt.internal.core.JavaModelManager$11.run(JavaModelManager.java:3195)\n    org.eclipse.jdt.internal.core.JavaModelManager.initializeAllContainers(JavaModelManager.java:3255)\n    org.eclipse.jdt.internal.core.JavaModelManager.getClasspathContainer(JavaModelManager.java:2209)\n    org.eclipse.jdt.core.JavaCore.getClasspathContainer(JavaCore.java:3939)\n    org.eclipse.jdt.internal.core.JavaProject.resolveClasspath(JavaProject.java:3159)\n    org.eclipse.jdt.internal.core.JavaProject.resolveClasspath(JavaProject.java:3323)\n    org.eclipse.jdt.internal.core.ClasspathChange.generateDelta(ClasspathChange.java:249)\n    org.eclipse.jdt.internal.core.DeltaProcessor.resourceChanged(DeltaProcessor.java:2134)\n    org.eclipse.jdt.internal.core.DeltaProcessingState.resourceChanged(DeltaProcessingState.java:503)\n    org.eclipse.core.internal.events.NotificationManager$1.run(NotificationManager.java:321)\n    org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)\n    org.eclipse.core.internal.events.NotificationManager.notify(NotificationManager.java:311)\n    org.eclipse.core.internal.events.NotificationManager.broadcastChanges(NotificationManager.java:174)\n    org.eclipse.core.internal.resources.Workspace.broadcastPostChange(Workspace.java:458)\n    org.eclipse.core.internal.resources.Workspace.endOperation(Workspace.java:1586)\n    org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2462)\n    org.eclipse.core.internal.events.NotificationManager$NotifyJob.run(NotificationManager.java:60)\n    org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)\n\nThreadId: 40 ThreadName: Worker-7 ThreadState: TIMED_WAITING\n  Blocked On: org.eclipse.core.internal.jobs.WorkerPool@41ce40fd LockOwnerId: -1 LockOwnerName: null\n  Synchronizers Locked: none\n  Monitors Locked: none\n  Stack Trace: \n    java.base@17.0.10/java.lang.Object.wait(Native Method)\n    org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)\n    org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)\n    org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)\n\tat org.eclipse.osgi.container.Module.lockStateChange(Module.java:370)\n\t... 23 more\n'
```

I have gotten nvim-jdtls itself working and wanted to scrap that and give this plugin a try. I may be doing something incorrectly so any help is greatly appreciated!

### Steps To Reproduce

1. Clone Kickstart.nvim
2. Add example installation and configuration (added "require('java').setup()" and "requrie('lspconfig').jdtls.setup({})" right before "require(mason-lspconfig).setup")
3. Open java file and see logs in LspLog

### Expected Behavior

JDTLS to start and provide LSP functionality

**Comment by exosyphon** [comment]
I updated my config and I believe I am getting a different error now.

```
[START][2024-03-18 09:03:39] LSP logging initiated
[ERROR][2024-03-18 09:03:39] .../vim/lsp/rpc.lua:734	"rpc"	"/usr/bin/java"	"stderr"	"Mar 18, 2024 9:03:39 AM org.apache.aries.spifly.BaseActivator log\nINFO: Registered provider ch.qos.logback.classic.servlet.LogbackServletContainerInitializer of service jakarta.servlet.ServletContainerInitializer in bundle ch.qos.logback.classic\n"
[ERROR][2024-03-18 09:03:39] .../vim/lsp/rpc.lua:734	"rpc"	"/usr/bin/java"	"stderr"	"Mar 18, 2024 9:03:39 AM org.apache.aries.spifly.BaseActivator log\nINFO: Registered provider ch.qos.logback.classic.spi.LogbackServiceProvider of service org.slf4j.spi.SLF4JServiceProvider in bundle ch.qos.logback.classic\n"
```

```
  {
    'nvim-java/nvim-java',
    dependencies = {
      'nvim-java/lua-async-await',
      'nvim-java/nvim-java-core',
      'nvim-java/nvim-java-test',
      'nvim-java/nvim-java-dap',
      'MunifTanjim/nui.nvim',
      'neovim/nvim-lspconfig',
      'mfussenegger/nvim-dap',
      {
        'williamboman/mason.nvim',
        opts = {
          registries = {
            'github:nvim-java/mason-registry',
            'github:mason-org/mason-registry',
          },
        },
      },
    },
    config = function()
      require('java').setup()
    end,
  },
```

and then added this line at the very bottom of the init.lua:

```
require('lspconfig').jdtls.setup {}
```

**Comment by s1n7ax** [comment]
@exosyphon I have no clue what that error is about. Classic java 12km long stack trace. What does your `Mason` UI shows? Can you send a screenshot

**Comment by s1n7ax** [comment]
@exosyphon It's fair if you gave up on neovim but please re-open if you want to give it another shot

**Comment by exosyphon** [comment]
Sorry, been traveling recently. I will get you a screenshot of the Mason UI by tomorrow. Is there anything you are looking for in particular?

**Comment by s1n7ax** [comment]
@exosyphon Mason packages installed and their versions

**Comment by exosyphon** [comment]
<img width="710" alt="Screenshot 2024-04-28 at 1 38 38 PM" src="https://github.com/nvim-java/nvim-java/assets/1758002/fcdf35d1-46aa-42ff-8724-fa0302cea828">

Here are the versions:
    ◍ java-debug-adapter - 0.55.0
    ◍ java-test - 0.40.1
    ◍ jdtls - 1.31.0
    ◍ kotlin-language-server - 1.3.9
    ◍ ktlint - 1.2.1
    ◍ lombok-nightly - nightly
    ◍ lua-language-server - 3.7.4
    ◍ openjdk-17 - 17.0.2
    ◍ stylua - 0.20.0

I have also published the branch here: https://github.com/exosyphon/KickstartNvim/tree/kotlin-config

**Comment by exosyphon** [comment]
Here is also what I see inside of a java file running :LspInfo

<img width="1376" alt="Screenshot 2024-04-28 at 1 44 24 PM" src="https://github.com/nvim-java/nvim-java/assets/1758002/7670009b-f733-488c-9c73-b14a4ee89a22">

**Comment by exosyphon** [comment]
Ok I am going to give this another go. Let me know whether I should open a new issue...

jdtls is still crashing on me. This is what I see in the lsp.log file.

```bash
[ERROR][2024-08-25 16:08:14] .../vim/lsp/rpc.lua:770	"rpc"	"/Users/andrew/.asdf/installs/java/temurin-21.0.4+7.0.LTS/bin/java"	"stderr"	"Error: Unable to initialize main class org.springframework.ide.vscode.boot.app.BootLanguageServerBootApp\nCaused by: java.lang.UnsatisfiedLinkError: 'void java.util.zip.Inflater.initIDs()'\n"
```

**Comment by timwenzeltech** [comment]
Does this problem exits with every java file, or only that specific (I am guessing springboot) project? If this project was initialized or created with vscode, you may want to look into the pom.xml main class spec.

**Comment by exosyphon** [comment]
Good question. I believe it is every Java file. It is a vanilla Java app that uses lombok. I generated it using `gradle init`. I will try generating a Spring Boot project using Spring Initializr and see if I get the same error.

**Comment by exosyphon** [comment]
Confirmed that the same issue is with Spring Boot app and any Java file. I think it has to do with setting the LD_LIBRARY_PATH, which I am unsure how to do on a Mac.

**Comment by timwenzeltech** [comment]
My knowledge on this nvim-java plugin is limited, you can take a look at my personal nvim repo, and look how I configured jdtls in my ftplugin/java.lua I made it work perfectly including debug and test. I took a look at your config and my personal recommendation would be to not rely on the lua functions to get directories like os.getenv("HOME") but instead write them out via "/home/user" or ~. Another thing is that your capabilities should also include standard cmp capabilities, again you may want to look at my config. I hope this is helpful to you!

**Comment by jugarpeupv** [comment]
This is happening to me, but only when i run  nvim-dap

It works on the first line, but the moment it goes into the System.out.print line it says the following error

<img width="1833" alt="image" src="https://github.com/user-attachments/assets/0bc27cc7-4242-40b9-9d16-fbf9035cda56">

The error:

Exception in thread "main" java.lang.UnsatisfiedLinkError: 'void java.util.zip.Inflater.initIDs()'
        at java.base/java.util.zip.Inflater.initIDs(Native Method)
        at java.base/java.util.zip.Inflater.<clinit>(Inflater.java:116)
        at java.base/java.util.zip.ZipFile$CleanableResource.getInflater(ZipFile.java:735)
        at java.base/java.util.zip.ZipFile$ZipFileInflaterInputStream.<init>(ZipFile.java:427)
        at java.base/java.util.zip.ZipFile.getInputStream(ZipFile.java:394)
        at java.base/java.util.jar.JarFile.getBytes(JarFile.java:800)
        at java.base/java.util.jar.JarFile.checkForSpecialAttributes(JarFile.java:1006)
        at java.base/java.util.jar.JarFile.isMultiRelease(JarFile.java:388)
        at java.base/java.util.jar.JarFile.getEntry(JarFile.java:510)
        at java.base/java.util.jar.JarFile.getJarEntry(JarFile.java:472)
        at java.base/jdk.internal.util.jar.JarIndex.getJarIndex(JarIndex.java:118)
        at java.base/jdk.internal.loader.URLClassPath$JarLoader$1.run(URLClassPath.java:770)
        at java.base/jdk.internal.loader.URLClassPath$JarLoader$1.run(URLClassPath.java:762)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
        at java.base/jdk.internal.loader.URLClassPath$JarLoader.ensureOpen(URLClassPath.java:761)
        at java.base/jdk.internal.loader.URLClassPath$JarLoader.<init>(URLClassPath.java:735)
        at java.base/jdk.internal.loader.URLClassPath$3.run(URLClassPath.java:499)
        at java.base/jdk.internal.loader.URLClassPath$3.run(URLClassPath.java:482)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
        at java.base/jdk.internal.loader.URLClassPath.getLoader(URLClassPath.java:481)
        at java.base/jdk.internal.loader.URLClassPath.getLoader(URLCla
        at java.base/jdk.internal.loader.URLClassPath.getResource(URLC
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassO
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassO
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.l
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:
        at com.example.demo.DemoApplication.main(DemoApplication.java:

[Process exited 1]


I believe this might be related to java.library.path??

https://rollbar.com/blog/java-unsatisfiedlinkerror-runtime-error/

Maybe @s1n7ax you know better?

I have a M1 macbook

My JAVA_HOME is https://rollbar.com/blog/java-unsatisfiedlinkerror-runtime-error/  



private/demo on  main [!?] via 🅶 v8.10.2 via ☕ v17.0.12 echo $JAVA_HOME
/Library/Java/JavaVirtualMachines/openjdk-17.jdk/Contents/Home

private/demo on  main [!?] via 🅶 v8.10.2 via ☕ v17.0.12 java --version
openjdk 17.0.2 2022-01-18
OpenJDK Runtime Environment (build 17.0.2+8-86)
OpenJDK 64-Bit Server VM (build 17.0.2+8-86, mixed mode, sharing)

private/demo on  main [!?] via 🅶 v8.10.2 via ☕ v17.0.12 java -XshowSettings:properties 
Property settings:
    file.encoding = UTF-8
    file.separator = /
    java.class.path = 
    java.class.version = 61.0
    java.home = /Users/jgarcia/.local/share/nvim/mason/packages/openjdk-17/jdk-17.0.2.jdk/Contents/Home
    java.io.tmpdir = /var/folders/ql/25x3bkjj0hd53qhj4x0hpngr0000gn/T/
    java.library.path = /opt/homebrew/lib
        /Users/jgarcia/Library/Java/Extensions
        /Library/Java/Extensions
        /Network/Library/Java/Extensions
        /System/Library/Java/Extensions
        /usr/lib/java
        .
    java.runtime.name = OpenJDK Runtime Environment
    java.runtime.version = 17.0.2+8-86
    java.specification.name = Java Platform API Specification
    java.specification.vendor = Oracle Corporation
    java.specification.version = 17
    java.vendor = Oracle Corporation
    java.vendor.url = https://java.oracle.com/
    java.vendor.url.bug = https://bugreport.java.com/bugreport/
    java.version = 17.0.2
    java.version.date = 2022-01-18
    java.vm.compressedOopsMode = Zero based
    java.vm.info = mixed mode, sharing
    java.vm.name = OpenJDK 64-Bit Server VM
    java.vm.specification.name = Java Virtual Machine Specification
    java.vm.specification.vendor = Oracle Corporation
    java.vm.specification.version = 17
    java.vm.vendor = Oracle Corporation
    java.vm.version = 17.0.2+8-86
    jdk.debug = release
    line.separator = \n 
    native.encoding = UTF-8
    os.arch = x86_64
    os.name = Mac OS X
    os.version = 15.0.1
    path.separator = :
    sun.arch.data.model = 64
    sun.boot.library.path = /Users/jgarcia/.local/share/nvim/mason/packages/openjdk-17/jdk-17.0.2.jdk/Contents/Home/lib
    sun.cpu.endian = little
    sun.io.unicode.encoding = UnicodeBig
    sun.java.launcher = SUN_STANDARD
    sun.jnu.encoding = UTF-8
    sun.management.compiler = HotSpot 64-Bit Tiered Compilers
    sun.stderr.encoding = UTF-8
    sun.stdout.encoding = UTF-8
    user.country = ES
    user.dir = /Users/jgarcia/private/demo
    user.home = /Users/jgarcia
    user.language = en
    user.name = jgarcia



Maybe java.library.path should include .local/share/nvim/mason/packages/openjdk-17/jdk-17.0.2.jdk/Contents/Home/lib ?? 

I see java.home pointing to   java.home = /Users/jgarcia/.local/share/nvim/mason/packages/openjdk-17/jdk-17.0.2.jdk/Contents/Home

but in the command line JAVA_HOME is pointing to /Library/Java/JavaVirtualMachines/openjdk-17.jdk/Contents/Home

**Comment by jugarpeupv** [comment]
Okey now i switched to java 23 and do not let Mason install the jdk and i still face the same issue with nvim-dap

I guess it is related to the dap configuration, i just read this from their repository, my guess is that the nvim-dap java configuration is not picking native java libraries @s1n7ax 

https://github.com/mfussenegger/nvim-dap/wiki/Java
